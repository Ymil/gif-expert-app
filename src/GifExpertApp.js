import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { ShowCategory } from './components/ShowCategory';

const GifExpertApp = ({defaultCategories = []}) => {
    
    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h1>Gif Expert App</h1>
            <AddCategory setCategories={setCategories} />
            <hr />            
            <ul>
                {
                    categories.map((category) => (
                        <ShowCategory 
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ul>
        </>
    )
}
export default GifExpertApp;