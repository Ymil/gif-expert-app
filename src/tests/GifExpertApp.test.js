import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"
describe('Probando GifExpertApp', () => {
    test('debe mostrar la interface', () => {        
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot()
    })

    test('debe mostrar una lista de categorias ', () => {
        const categories = [
            'Futbol', 'Basket', 'Ciclismo'
        ]

        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot()

        expect(wrapper.find('ShowCategory').length).toBe(categories.length)

    })
    
    
})
