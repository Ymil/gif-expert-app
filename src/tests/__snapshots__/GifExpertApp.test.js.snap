// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifExpertApp debe mostrar la interface 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Probando GifExpertApp debe mostrar una lista de categorias  1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <ShowCategory
      category="Futbol"
      key="Futbol"
    />
    <ShowCategory
      category="Basket"
      key="Basket"
    />
    <ShowCategory
      category="Ciclismo"
      key="Ciclismo"
    />
  </ul>
</Fragment>
`;
